core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1345,7 +1380,7 @@ private void replaceOrUpsertOneFromResolution(
       final MongoNamespace namespace,
       final BsonValue documentId,
       final BsonDocument document,
-      final BsonValue atVersion,
+      final BsonDocument atVersion,
```

Comment:
This is because the consumer of `atVersion` is only operating on the pure BSON of the version, whereas `DocumentVersionInfo` is an abstraction on top of that that we use for various other checks.

`DocumentVersionInfo` is allowed to represent an empty version, meaning consumers of atVersion would then have to check for those kinds of states. I think this is fine for now, especially since consumers of `atVersion` never convert back to a `DocumentVersionInfo`, and they don't have any use for its methods/functionality

Reply:
👍